android-pay/src/main/java/com/stripe/wrap/pay/activity/StripeAndroidPayActivity.java

```diff
@@ -0,0 +1,553 @@
+        // This is a good place to display a spinner if you anticipate network delays
+        // initializing the Google API client.
+    }
+
+    protected void onAfterAndroidPayCheckComplete() {
+        // If a spinner was showing, remove it in this method.
+    }
+
+    /**
+     * Override to handle Google errors in custom ways.
+     *
+     * @param errorCode the error code returned from the {@link GoogleApiClient}
+     */
+    protected void handleError(int errorCode) { }
+
+    /**
+     * Override this method to display the Android Pay confirmation wallet fragment. Place
+     * it in a container of your choice using a {@link android.support.v4.app.FragmentTransaction}.
+     *
+     * @param walletFragment a {@link SupportWalletFragment} created using the
+     */
+    protected void addConfirmationWalletFragment(@NonNull SupportWalletFragment walletFragment) { }
+
+    /**
+     * Override this method to display the Android Pay fragment (a clickable button). Place
+     * it in a container of your choice using a {@link android.support.v4.app.FragmentTransaction}.
+     *
+     * @param walletFragment a {@link SupportWalletFragment} created using the
+     */
+    protected void addBuyButtonWalletFragment(@NonNull SupportWalletFragment walletFragment) { }
+
+    /**
+     * Override this method to react to a {@link MaskedWallet} being returned from the Google API
+     * when the user is asked to confirm wallet choices.
+     *
+     * @param maskedWallet the {@link MaskedWallet} returned from the {@link GoogleApiClient}
+     */
+    protected void onConfirmedMaskedWalletRetrieved(@Nullable MaskedWallet maskedWallet) { }
+
+    /**
+     * Override this method to react to a {@link MaskedWallet} being returned from
+     * the Google API. The masked wallet will have the user's shipping information (if
+     * it was required), which you can use to update the final shipping price. You can also
+     * use the card information as a confirmation for the user.
+     *
+     * @param maskedWallet the {@link MaskedWallet} returned from the {@link GoogleApiClient}
+     */
+    protected void onMaskedWalletRetrieved(@Nullable MaskedWallet maskedWallet) { }
+
+    /**
+     * Override this function to move to {@link WalletConstants#ENVIRONMENT_PRODUCTION}
+     *
+     * @return the current wallet environment
+     */
+    protected int getWalletEnvironment() {
+        return WalletConstants.ENVIRONMENT_TEST;
+    }
+
+    /**
+     * Override this function to change the theme of Wallet display items
+     *
+     * @return the current wallet theme
+     */
+    protected int getWalletTheme() {
+        return WalletConstants.THEME_LIGHT;
+    }
+
+    /**
+     * Called when a Stripe {@link Source} is returned from Google's servers.
+     * Send the token in this method to your server to make a charge.
+     *
+     * Note: this feature is not yet implemented, but to be future-proof, a returned
```

Comment:
hmm, requiring that people implement both of these callbacks kind of complicates the API. maybe we could make Token and Source conform to the same interface, and just have one callback? On iOS our token/source interface just has an ID, which is all you need to send to your server.
https://github.com/stripe/stripe-ios/blob/master/Stripe/PublicHeaders/STPSourceProtocol.h